Update and simplify COPYING.txt (#16830)
nvaccess/nvda: projectDocs/dev/copyrightHeaders.md

> In NVDA files, which are licensed under the GNU GPL, use the following copyright header:

Reviewer: But their not, are they?

```suggestion
In NVDA files, which are licensed under the terms of a modified GNU General Public License, use the following copyright header:
```

> ```

Reviewer 2: ```suggestion
```py
```

# A part of NonVisual Desktop Access (NVDA)
# Copyright (C) <CREATED YEAR>-<LAST UPDATED YEAR> NV Access Limited, <YOUR NAME>

> # This file may be used under the terms of the GNU General Public License, version 2 or later.

Reviewer: Again, I thought we were operating under a modified license? If so, the files should have their headers indicate the project license, as a derivative of the GPL V2, but shouldn't say that they are licensed under GPL V2--else the modified project license has no meaning.

```suggestion
# This file may be used under the terms of the GNU General Public License, version 2 or later, as modified by the NVDA license.
```

Author: I can see the argument of doing it either way. What I'm aiming to do is to simplify as much as possible: since the modifications to GPL-2 are in two very specific areas: "Non-GPL Components in Plugins and Drivers" and "Microsoft Distributable Code", then all code using this header will not be in either of those two areas and will therefore be GPL2.

Author: If this assumption is incorrect, please let me know @XLTechie

> # For more details see: https://www.gnu.org/licenses/gpl-2.0.html

Reviewer: Shouldn't this be a link to NVDA's license file, since it overrides GPL?

```

Replacing:
- `<YOUR NAME>` with your name.
- `<CREATED YEAR>` the year the file was created

> - `<LAST UPDATED YEAR>`when the file was last updated. For new files this can be missing. E.g. a file created in 2025 might have `#Copyright (C) 2025`

Reviewer 2: ```suggestion
- `<LAST UPDATED YEAR>` when the file was last updated.
For new files this can be missing.
e.g. a file created in 2025 might have `# Copyright (C) 2025`
```

```

> Some files are covered by the GNU LGPL, for example the the controller client and its examples. This allows someone to use them (or parts of them) as-is. In this case use the following header:

Reviewer 2: ```suggestion
Some files are covered by the GNU LGPL, for example the the controller client and its examples.
This allows someone to use them (or parts of them) as-is.
In this case use the following header:
```

Reviewer: @seanbudd, in your suggestion, can you fix the double "the" in the first line as well?  "`for example the the controller client`"

> ```

Reviewer 2: ```suggestion
```py
```

# A part of NonVisual Desktop Access (NVDA)
# Copyright (C) <CREATED YEAR>-<LAST UPDATED YEAR> NV Access Limited, <YOUR NAME>
# This file may be used under the terms of the GNU Lesser General Public License, version 2.1 or later.
# For more details see: https://www.gnu.org/licenses/lgpl-2.1.html
```

> In some files an older style of referring to the contributors is used. The contributors is not a list of names and may just say something like:

Reviewer 2: ```suggestion
In some files an older style of referring to the contributors is used.
The contributors is not a list of names and may just say something like:
```

> `Copyright (C) 2006-2024 NVDA Contributors`. We suggesting referring to the git logs to identify specific authors.

Reviewer 2: ```suggestion
`Copyright (C) 2006-2024 NVDA Contributors`.
We suggesting referring to the git logs to identify specific authors.
```

Reviewer: "we suggesting" s/b "we suggest".

> `Copyright (C) 2006-2024 NVDA Contributors`. We suggesting referring to the git logs to identify specific authors.

Reviewer: Fix "suggesting".

```suggestion
`Copyright (C) 2006-2024 NVDA Contributors`.
We suggest referring to the git logs to identify specific authors.
```

In NVDA files, which are licensed under the terms of a modified GNU General Public License, use the following copyright header:

```py
# A part of NonVisual Desktop Access (NVDA)
# Copyright (C) <CREATED YEAR>-<LAST UPDATED YEAR> NV Access Limited, <YOUR NAME>
# This file may be used under the terms of the GNU General Public License, version 2 or later, as modified by the NVDA license.
# For full terms and any additional permissions, see the NVDA license file: https://github.com/nvaccess/nvda/blob/master/copying.txt
```

Replacing:
- `<YOUR NAME>` with your name.
- `<CREATED YEAR>` the year the file was created
- `<LAST UPDATED YEAR>` when the file was last updated.
For new files this can be missing.
e.g. a file created in 2025 might have `# Copyright (C) 2025`

> ```

Reviewer 2: ```suggestion
```
